### PR TITLE
Orca: add pytorch resume training to NCF example

### PR DESCRIPTION
**Pull request by zpeng1898**

### CODE REVIEW

File: python/orca/tutorial/NCF/pytorch_resume_xshards.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+
+# Step 1: Init Orca Context
+args = parse_args("PyTorch NCF Resume Training with Orca Xshards")
+init_orca(args, extra_python_lib="pytorch_model.py")
+
+
+# Step 2: Load the processed data
+train_data = XShards().load_pickle(os.path.join(args.data_dir, "train_processed_xshards"))
+test_data = XShards().load_pickle(os.path.join(args.data_dir, "test_processed_xshards"))
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_resume_xshards.py:38]: XShards() -> XShards

File: python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    return model
+
+
+def optimizer_creator(model, config):
+    return optim.Adam(model.parameters(), lr=config["lr"])
+
+
+def scheduler_creator(optimizer, config):
+    scheduler = StepLR(optimizer, step_size=1)
+    return scheduler
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py:62]: change this and remove the import same as in train.py

File: python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+from utils import *
+
+from bigdl.orca.learn.pytorch import Estimator
+from bigdl.orca.learn.pytorch.callbacks.tensorboard import TensorBoardCallback
+from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
+
+
+# Step 1: Init Orca Context
+args = parse_args("PyTorch NCF Resume Training with Spark DataFrame")
+init_orca(args, extra_python_lib="pytorch_model.py")
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py:32]: change to args.cluster_mode

File: python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+
+# Step 3: Define the model, optimizer and loss
+def model_creator(config):
+    model = NCF(user_num=config["user_num"],
+                item_num=config["item_num"],
+                factor_num=config["factor_num"],
+                num_layers=config["num_layers"],
+                dropout=config["dropout"],
+                model=config["model"],
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py:48]: use the default value for model

File: python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+est = Estimator.from_torch(model=model_creator,
+                           optimizer=optimizer_creator,
+                           scheduler_creator=scheduler,
+                           loss=loss,
+                           metrics=[Accuracy(), Precision(), Recall()],
+                           backend=args.backend,
+                           use_tqdm=True,
+                           workers_per_node=args.workers_per_node,
+                           config=config)
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py:81]: make the order align with train

File: python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+est.load(os.path.join(args.model_dir, "NCF_model"))
+train_stats = est.fit(data=train_df, epochs=2,
+                      feature_cols=feature_cols,
+                      label_cols=label_cols,
+                      batch_size=10240,
+                      validation_data=test_df,
+                      callbacks=callbacks)
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py:88]: format this same as train

File: python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+
+# Step 5: Distributed evaluation of the trained model
+eval_stats = est.evaluate(data=test_df,
+                          feature_cols=feature_cols,
+                          label_cols=label_cols,
+                          batch_size=10240)
+print("Evaluation results:")
+for k, v in eval_stats.items():
+    print("{}: {}".format(k, v))
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py:103]: For resume training, I think we can remove the evaluate here, to make it simple.

**Comment by hkvision** [comment]
rename file names to `...resume_train...`

File: python/orca/tutorial/NCF/pytorch_resume_train_dataloader.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+
+# Step 1: Init Orca Context
+args = parse_args("PyTorch NCF Resume Training with DataLoader")
+init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py,pytorch_dataset.py")
+
+
+# Step 2: Define train and test datasets as PyTorch DataLoader
+def train_loader_func(config, batch_size):
+    train_dataset, _ = load_dataset(config["data_dir"], num_ng=4)
```
Comment (hamham223) [inline python/orca/tutorial/NCF/pytorch_resume_train_dataloader.py:38]: Seems this should be `config["dataset_dir"]`? It's not consist with https://github.com/intel-analytics/BigDL/blob/main/python/orca/tutorial/NCF/pytorch_train_dataloader.py#L64

Here's what's in my local `config.json`
```json
{"dataset_dir": "./ml-1m", "factor_num": 16, "num_layers": 3, "dropout": 0.5, "lr": 0.01, "sparse_feats_embed_dims": 8}
```

**Pull request merged**